lang/test/org/partiql/lang/eval/like/PatternTests.kt

```diff
@@ -151,32 +151,60 @@ class PatternPartTests {
+        val actualMatches = pat.matcher(tc.input).matches()
 
         Assert.assertEquals(tc.shouldMatch, actualMatches)
     }
 
     @Test
     fun patternParserTest() {
         // the parser should consider multiple consecutive % to be the same as one
-        val patParts = parsePattern("%%a%%%_%%% %%", escapeChar = null)
-        assertEquals(
-            listOf(
-                PatternPart.ZeroOrMoreOfAnyChar,
-                PatternPart.ExactChars("a".codePoints().toArray()),
-                PatternPart.ZeroOrMoreOfAnyChar,
-                PatternPart.AnyOneChar,
-                PatternPart.ZeroOrMoreOfAnyChar,
-                PatternPart.ExactChars(" ".codePoints().toArray()),
-                PatternPart.ZeroOrMoreOfAnyChar
-            ),
-            patParts
-        )
+        val pattern = parsePattern("%%a%%%_%%% %%", escapeChar = null)
+        assertEquals("^.*?\\Qa\\E.+?\\Q \\E.*?$", pattern.pattern())
     }
 
     @Test
     fun stressTest() {
         // makes absolutely certain we do not stack overflow on too many consecutive `%` characters
-        assertEquals(true, executePattern(parsePattern("%".repeat(10000) + "a", escapeChar = null), "aaaa"))
+        assertEquals(true, parsePattern("%".repeat(10000) + "a", escapeChar = null).matcher("aaaa").matches())
+    }
+
+    @Test
+    fun like() {
+        assertEquals("^\\Qfoo\\E$", parsePattern("foo", '\\'.toInt()).pattern())
+
+        assertEquals("^.*?\\Qfoo\\E$", parsePattern("%foo", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.*?$", parsePattern("foo%", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.*?\\Qbar\\E$", parsePattern("foo%bar", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.*?\\Qbar\\E$", parsePattern("foo%%bar", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.*?\\Qbar\\E$", parsePattern("foo%%%bar", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.*?\\Qbar\\E$", parsePattern("foo%%%%bar", '\\'.toInt()).pattern())
+        assertEquals(
+            "^.*?\\Qfoo\\E.*?\\Qbar\\E.*?$",
+            parsePattern("%foo%%%%bar%", '\\'.toInt()).pattern(),
+        )
+        assertEquals(
+            "^.*?\\Qfoo\\E.*?\\Qbar%baz\\E.*?$",
+            parsePattern("%foo%%%%bar\\%baz%", '\\'.toInt()).pattern(),
+        )
+        assertEquals(
+            "^.*?\\Qfoo\\E.*?\\Qbar%baz\\E.*?$",
+            parsePattern("%foo%%%%bar*%baz%", '*'.toInt()).pattern(),
+        )
+        assertEquals("^.\\Qfoo\\E$", parsePattern("_foo", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.$", parsePattern("foo_", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.\\Qbar\\E$", parsePattern("foo_bar", '\\'.toInt()).pattern())
+        assertEquals("^\\Qfoo\\E.{2,2}\\Qbar\\E$", parsePattern("foo__bar", '\\'.toInt()).pattern())
+        assertEquals(
+            "^\\Qfoo\\E.\\Q.*?\\E.\\Qbar\\E$",
+            parsePattern("foo_.*?_bar", '\\'.toInt()).pattern(),
+        )
```

Comment:
Could also add a test for a pattern with at least two of a character (i.e. '_'). Example query: `foo_%_bar`, `foo_%_%bar`, or `foo%_%%_%bar`